R/oai_available.R

```diff
@@ -0,0 +1,36 @@
+#' Test of OAI-PMH service is available
+#'
+#' Silently test if OAI-PMH service is available under the URL provided.
+#'
+#' @param u base URL to OAI-PMH service
+#' @param ... other arguments passed to \code{\link{id}}
+#'
+#' @return \code{TRUE} or \code{FALSE} if the service is available.
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' url_list <- list(
+#'   archivesic="http://archivesic.ccsd.cnrs.fr/oai/oai.php",
+#'   datacite = "http://oai.datacite.org/oai",
+#'
+#'   # No OAI-PMH here
+#'   google = "http://google.com"
+#' )
+#'
+#' sapply(url_list, oai_available)
+#' }
+oai_available <- function( u, ... ) {
+  # Identify yourself!
+  r <- try( id(u, ...), silent=TRUE )
+  if( inherits(r, "try-error") ) return(FALSE)
```

Comment:
I guess it could... :D Initially I wrote something more complicated (before I found `id()`) and edited it out. BTW what's with the codecov checks? I did not know it became a separate service.